services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx

```diff
@@ -1,26 +1,70 @@
-import React, { useEffect } from 'react'
-import { GridContainer, Link } from '@trussworks/react-uswds'
-import { NavLink, useParams, useLocation } from 'react-router-dom'
+import { Alert, Button, ButtonGroup, GridContainer, Link, Modal, ModalFooter, ModalHeading, ModalRef, ModalToggleButton, Textarea } from '@trussworks/react-uswds'
+import { GraphQLError } from 'graphql'
+import React, { useEffect, useRef, useState } from 'react'
+import { NavLink, useLocation, useParams } from 'react-router-dom'
 import sprite from 'uswds/src/img/sprite.svg'
-import styles from './SubmissionSummary.module.scss'
-import { useFetchStateSubmissionQuery } from '../../gen/gqlClient'
 import { Loading } from '../../components/Loading'
 import {
-    SubmissionTypeSummarySection,
-    ContractDetailsSummarySection,
-    RateDetailsSummarySection,
-    ContactsSummarySection,
-    SupportingDocumentsSummarySection,
+    ContactsSummarySection, ContractDetailsSummarySection,
+    RateDetailsSummarySection, SubmissionTypeSummarySection, SupportingDocumentsSummarySection
 } from '../../components/SubmissionSummarySection'
-import { GenericError } from '../Errors/GenericError'
-import { usePage } from '../../contexts/PageContext'
 import { useAuth } from '../../contexts/AuthContext'
+import { usePage } from '../../contexts/PageContext'
+import { DraftSubmission, UnlockStateSubmissionMutationFn, useFetchStateSubmissionQuery, useUnlockStateSubmissionMutation } from '../../gen/gqlClient'
+import { GenericError } from '../Errors/GenericError'
+import styles from './SubmissionSummary.module.scss'
+
+function unlockModalButton(modalRef: React.RefObject<ModalRef>) {
+    return (
+        <ModalToggleButton
+            modalRef={modalRef}
+            data-testid="form-submit"
+            outline
+            opener
+        >
+            Unlock submission
+        </ModalToggleButton>
+        )
+}
+
+// This wrapper gets us some reasonable errors out of our unlock call. This would be a good candidate
+// for a more general and generic function so that we can get more sensible errors out of all of the 
+// generated mutations.
+async function unlockMutationWrapper(unlockStateSubmission: UnlockStateSubmissionMutationFn, id: string): Promise<DraftSubmission | readonly GraphQLError[] | Error> {
+    try {
+            const result = await unlockStateSubmission({
+                variables: {
+                    input: {
+                        submissionID: id
+                    }
+                }
+            })
+
+            if (result.errors) {
+                return result.errors
```

Comment:
Yes we need something like this. And  me this should log the errors and then either throw a `Error` or return a error type. `GraphQLError`  may be something we need looks like. 

An array is not a clear error type, especially on the frontend where often the result is an array of data we iterate over.